### PR TITLE
Shell script json inteface

### PR DESCRIPTION
**Pull request by iamveronika**
Currently the shell script works as:
```
julia --project=. runfit.jl -p model_spec.json
```

### CODE REVIEW

**Comment by cortner** [comment]
Thank you. I'll review asap.

File: Project.toml
Comment (cortner) [inline Project.toml]: why is JSON3 back, I thought we deleted this?

File: src/json_interface.jl
Comment (cortner) [inline src/json_interface.jl]: This doesn't look right. I think it should be read from the current project. Something like this: 
```julia-repl
shell> pwd
/Users/ortner/gits/ACEpotentials.jl/docs/src/tutorials

julia> using Pkg

julia> p = Pkg.project()
Pkg.API.ProjectInfo("ACEpotentials", UUID("3b96b61c-0fcc-4693-95ed-1ef9f35fcc53"), v"0.8.0-dev", true, Dict{String, Base.UUID}("Pkg" => UUID("44cfe95a-1eb2-52ea-b672-e2afdf69b78f"), "ForwardDiff" => UUID("f6369f11-7733-5829-9624-2563aa707210"), "SpheriCart" => UUID("5caf2b29-02d9-47a3-9434-5931c85ba645"), "Random" => UUID("9a3f8284-a2c9-5f02-9a11-845980a1fd5c"), "Polynomials4ML" => UUID("03c4bcba-a943-47e9-bfa1-b1661fc2974f"), "RepLieGroups" => UUID("f07d36f2-91c4-427a-b67b-965fe5ebe1d2"), "Interpolations" => UUID("a98d9a8b-a2ab-59e6-89dd-64a1c18fca59"), "ACEfit" => UUID("ad31a8ef-59f5-4a01-b543-a85c2f73e95c"), "AtomsCalculators" => UUID("a3e0e189-c65a-42c1-833c-339540406eb1"), "Optim" => UUID("429524aa-4258-5aef-a3af-852621145aeb")…), "/Users/ortner/gits/ACEpotentials.jl/Project.toml")

julia> p.
dependencies  ispackage
name          path
uuid          version
julia> p.path
"/Users/ortner/gits/ACEpotentials.jl/Project.toml"
```

Comment (cortner) [reply src/json_interface.jl]: In fact, this suggests to load and store both `Project.toml` and `Manifest.toml`.

Comment (cortner) [reply src/json_interface.jl]: something like this gets you the manifest path: 
```
manifest_path = p.path[1:end-12] * "Manifest.toml"
```
but it feels like a hack. Julia should have a better way to remove the filename from the string.

Comment (cortner) [reply src/json_interface.jl]: And here it is: 
```
joinpath(splitpath(p.path)[1:end-1]...)
```

File: src/json_interface.jl
Comment (cortner) [inline src/json_interface.jl:114]: as a general rule you should try to avoid introducing white-space changes. This makes it very hard for me to review and see what you have changed and what has remained the same.

File: src/json_interface.jl
Comment (cortner) [inline src/json_interface.jl:150]: please add a `save_project = true` flag to the keyword arguments.

File: scripts/runfit.jl
Comment (cortner) [inline scripts/runfit.jl]: please avoid all whitespace changes that don't achieve anything, unless you are the maintainer of a repository.

File: src/json_interface.jl
Comment (cortner) [reply src/json_interface.jl:150]: If the flag is true (default) then read both project and manifest and store them. If false, then don't do this.

**Review by cortner — CHANGES_REQUESTED** [review]
thank you, good draft. Can you please look at my suggested changes though before we merge?

File: Project.toml
Comment (iamveronika) [reply Project.toml]: Yes, I've used it, but I'll delete it

File: src/json_interface.jl
Comment (iamveronika) [reply src/json_interface.jl]: Done

File: src/json_interface.jl
Comment (iamveronika) [reply src/json_interface.jl:114]: Sorry for that, I'll avoid these changes

File: src/json_interface.jl
Comment (iamveronika) [reply src/json_interface.jl:150]: Done

**Comment by cortner** [comment]
can you try two more things please? 
- merge `main` into your branch and resolve conflicts so it can be merged
- see if you can write functionality to recover the project and manifest

**Comment by CheukHinHoJerry** [comment]
I can help a bit if @iamveronika haven’t write anything on this further yet?

**Comment by cortner** [comment]
That would be helpful. I can move the PR to the branch inside of ACEsuit.

**Pull request merged**
